api/ask_astro/slack/controllers/feedback/bad.py

```diff
@@ -10,54 +12,81 @@
     try:
-        user = body["user"]["id"]
-        channel = body["channel"]["id"]
-        thread_ts = body["message"]["thread_ts"]
-        message_ts = body["message"]["ts"]
-        value = body["actions"][0]["value"]
+        return {
+            "user": body["user"]["id"],
+            "channel": body["channel"]["id"],
+            "thread_ts": body["message"]["thread_ts"],
+            "message_ts": body["message"]["ts"],
+            "value": body["actions"][0]["value"],
+        }
     except KeyError as e:
-        logger.error(f"Missing key: {e}")
-        return
+        logger.error("Missing key: %s", e)
+        return {}
 
-    request_id = value.split(":")[0]
 
-    await submit_feedback(request_id, False, source_info={"type": "slack", "user": user})
+async def handle_feedback_bad(
+    body: dict[str, Any], ack: AsyncAck, respond: AsyncRespond, client: AsyncWebClient
+) -> None:
+    """
+    Handle feedback received from Slack and send appropriate responses.
+
+    :param body: The slack event body.
+    :param ack: Acknowledgement object from slack_bolt.
+    :param respond: Response object from slack_bolt.
+    :param client: Slack API client.
+    """
+    await ack()
+
+    details = extract_feedback_details(body)
+    if not details:
+        return
+
+    request_id = details["value"].split(":")[0]
+    await submit_feedback(request_id, False, source_info={"type": "slack", "user": details["user"]})
 
     async with TaskGroup() as tg:
-        tg.create_task(
-            respond(
-                f"😥 Thank you for your feedback, <@{user}>!",
-                thread_ts=thread_ts,
-                replace_original=False,
-                response_type="ephemeral",
-            )
-        )
-
-        async def update_reaction():
-            try:
-                await client.reactions_add(
-                    name="thumbsdown",
-                    channel=channel,
-                    timestamp=message_ts,
-                )
-            except SlackApiError as e:
-                # ignore the error if the reaction already exists
-                if e.response["error"] != "already_reacted":
-                    raise e
-
-            try:
-                await client.reactions_remove(
-                    name="thumbsup",
-                    channel=channel,
-                    timestamp=message_ts,
-                )
-            except Exception:
-                pass
-
-        tg.create_task(
-            update_reaction(),
-        )
+        tg.create_task(_send_response(details, respond))
+        tg.create_task(_update_reaction(details, client))
+
+
+async def _send_response(details: dict[str, str], respond: AsyncRespond) -> None:
+    """
+    Send a response back to the user in Slack.
+
+    :param details: The details extracted from the Slack body.
+    :param respond: Response object from slack_bolt.
+    """
+    await respond(
+        "😥 Thank you for your feedback, <@%s>!" % details["user"],
```

Comment:
```suggestion
        "😥 Thank you for your feedback, <@{details['user']}>!",
```

This is not logging. I think we're good to use f-string

Reply:
changed